doc/tutorial/text_analytics/data/movie_reviews/fetch_data.py

```diff
@@ -20,8 +20,15 @@
     if not os.path.exists(ARCHIVE_NAME):
         print("Downloading dataset from %s (3 MB)" % URL)
         opener = urlopen(URL)
-        open(ARCHIVE_NAME, 'wb').write(opener.read())
+        with open(ARCHIVE_NAME, 'wb') as archive:
+            archive.write(opener.read())
 
     print("Decompressing %s" % ARCHIVE_NAME)
-    tarfile.open(ARCHIVE_NAME, "r:gz").extractall(path='.')
+    try:
+        with tarfile.open(ARCHIVE_NAME, "r:gz") as archive:
+            archive.extractall(path = '.')
+    except AttributeError: # In Python 2.6, tarfile did not yet implement the context manager protocol
```

Comment:
I think this can be written:

```python
from contextlib import closing

with closing(tarfile.open(ARCHIVE_NAME, "r:gz")) as archive:
    archive.extractall(path = '.')
```

and that should work with any version of Python supported by sklearn.